### ISSUE TITLE
NoMethodError - undefined method `include?' for nil:NilClass

### ISSUE THREAD

**Issue by neverabe** [CocoaPods/CocoaPods#3797]
MacBook-Pro-2:Source user$ pod install
Analyzing dependencies

CocoaPods 0.38.0.beta.2 is available.
To update use: `gem install cocoapods --pre`
[!] This is a test version we'd love you to try.

For more information see http://blog.cocoapods.org
and the CHANGELOG for this version http://git.io/BaH8pQ.

――― MARKDOWN TEMPLATE ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

### Command

```
/usr/bin/pod install
```

### Report

* What did you do?

* What did you expect to happen?

* What happened instead?

### Stack

```
   CocoaPods : 0.37.2
        Ruby : ruby 2.0.0p481 (2014-05-08 revision 45883) [universal.x86_64-darwin14]
    RubyGems : 2.0.14
        Host : Mac OS X 10.10.3 (14D136)
       Xcode : 6.3.2 (6D2105)
         Git : git version 2.3.2 (Apple Git-55)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 2b1515b328fc913e47c05aa7912157b00666d925
```

### Plugins

```
cocoapods-plugins : 0.4.2
cocoapods-trunk   : 0.6.1
cocoapods-try     : 0.4.5
```

### Podfile

```ruby
# Uncomment this line to define a global platform for your project
# platform :ios, '6.0'

target 'GDataFramework' do

end

target 'GDataUnitTests' do

end

target 'DevelopmentTestTool' do

end
pod 'Google-API-Client/Drive', '~> 1.0'
```

### Error

```
NoMethodError - undefined method `include?' for nil:NilClass
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-0.24.3/lib/xcodeproj/project/object/native_target.rb:92:in `platform_name'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer/analyzer.rb:661:in `block in compute_platform_for_target_definition'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer/analyzer.rb:660:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer/analyzer.rb:660:in `compute_platform_for_target_definition'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer/analyzer.rb:709:in `block (2 levels) in compute_target_platforms'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer/analyzer.rb:704:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer/analyzer.rb:704:in `block in compute_target_platforms'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/user_interface.rb:59:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer/analyzer.rb:703:in `compute_target_platforms'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer/analyzer.rb:57:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer.rb:196:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer.rb:114:in `block in resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/user_interface.rb:59:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer.rb:113:in `resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer.rb:93:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/command/project.rb:71:in `run_install_with_update'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/command/project.rb:101:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-0.8.2/lib/claide/command.rb:312:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/command.rb:46:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/bin/pod:44:in `<top (required)>'
/usr/bin/pod:23:in `load'
/usr/bin/pod:23:in `<main>'
```

――― TEMPLATE END ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

[!] Oh no, an error occurred.

Search for existing GitHub issues similar to yours:
https://github.com/CocoaPods/CocoaPods/search?q=undefined+method+%60include%3F%27+for+nil%3ANilClass&type=Issues

If none exists, create a ticket, with the template displayed above, on:
https://github.com/CocoaPods/CocoaPods/issues/new

Be sure to first read the contributing guide for details on how to properly submit a ticket:
https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md

Don't forget to anonymize any private data!

**Comment by amorde** [comment]
It looks like `cocoapods` and/or `xcodeproj` cannot determine the platform of your project (either ios, osx, or watchOS).

Although you shouldn't have to, try adding `platform :ios`  or `platform :ios, '8.0'` to the top of your podfile and see if that fixes the issue.

**Comment by samsonjs** [comment]
That fixed it for me.

**Comment by segiddins** [comment]
Yeah, think the issue there is the empty target blocks?

**Comment by amorde** [comment]
Interesting. Since `pod init` puts two empty targets into the Podfile (one of which, the test target, usually remains empty in early development) this should probably be fixed. I'm not familiar with this area of the codebase though.

**Comment by samsonjs** [comment]
I should elaborate on my Podfile, which has no empty target blocks. We recently extracted our config from Xcode project file to xcconfig files. After that xcproj was unable to determine our platform automatically.

The Podfile looked roughly like this before I added a platform:

```ruby
source 'https://github.com/CocoaPods/Specs.git'
inhibit_all_warnings!

target :Blah do
  pod 'Tweaks'
  # etc.
end

target :BlahTests do
  pod 'OHHTTPStubs'
  pod 'OCMock'
end
```

**Comment by segiddins** [comment]
Ah, in that case could you share a project that reproduces the issue, @samsonjs ?

**Comment by matrinox** [comment]
You need to set the base SDK for the project. Inspecting the code shows:

```ruby
def platform_name
  if sdk.include? 'iphoneos'
    :ios
  …
```
And the definition of `sdk`:
```ruby
common_resolved_build_setting('SDKROOT')
```

Maybe an error should be thrown that tells the user to set the base SDK if `sdk` is evaluated to nil

**Comment by Spencevail** [comment]
Getting the same issue, here's my Podfile:

```ruby
platform :ios, '8.0'

use_frameworks!

pod 'HockeySDK',                        '~> 3.6.4'
pod 'Cartography',                      '~> 0.5.0'
pod 'XCGLogger',                        '~> 2.1.1'
pod 'ReactKit',                         '~> 0.11.0'
pod 'Argo',                             '~> 1.0.0'
pod 'MagicalRecord',                    '~> 2.3.0'
pod 'Result',                           '~> 0.4.0'

# Debuging utilities
pod 'Reveal-iOS-SDK',                   '~> 1.5.1', :configurations => ['Staging-Debug']

target 'InspiratoTests', :exclusive => true do
    pod 'Nimble',                       '~> 0.4.0'
    pod 'OHHTTPStubs',                  '~> 4.0.1'
end
```

Going back to Cococapods 0.37.2 fixes the issue for me.

**Comment by kreeger** [comment]
I'm with @spencevail here — I've got very similarly structured Podfile, with a `platform` declaration on the top of my Podfile, but v0.38.0 crashes whereas 0.37.2 does not.

**Comment by segiddins** [comment]
Still need a project that demonstrates this issue to investigate...

**Comment by matrinox** [comment]
https://github.com/matrinox/CocoaPodsMissingPlatform

**Comment by segiddins** [comment]
This appears to be fixed in Xcodeproj master.

**Comment by chadpav** [comment]
I'm seeing this exact error w/ cocoa pods 1.0 rc1.  I don't see it if I roll back to 0.39 (I apparently need 1.0 to address a different issue w/ Bolts and non-modular frameworks).  

I'm kind of stuck, what do you mean it was fixed in Xcodeproj master?


### Command

```
/usr/local/bin/pod install
```

### Report

* What did you do?

* What did you expect to happen?

* What happened instead?


### Stack

```
   CocoaPods : 1.0.0.rc.1
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin15]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.11.4 (15E65)
       Xcode : 7.3 (7D175)
         Git : git version 2.6.4 (Apple Git-63)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ b417248a65289d94f530ba6cbcae4f224606e527
```

### Plugins

```
cocoapods-deintegrate : 1.0.0.rc.1
cocoapods-plugins     : 1.0.0.rc.1
cocoapods-search      : 1.0.0.rc.1
cocoapods-stats       : 1.0.0.rc.1
cocoapods-trunk       : 1.0.0.rc.1
cocoapods-try         : 1.0.0.rc.1
```

### Podfile

```ruby
source 'https://github.com/CocoaPods/Specs.git'
platform :ios, '8.0' # This setting prevents a compiler warning in the ParseUI pod
use_frameworks!

pod 'SDWebImage',               '3.7.4'   # https://github.com/rs/SDWebImage/releases
pod 'Alamofire',                '3.1.5'   # https://github.com/Alamofire/Alamofire/releases
pod 'DZNEmptyDataSet',          '1.7.3'   # https://github.com/dzenbot/DZNEmptyDataSet/releases
pod 'Fabric',                   '1.6.6'   # https://docs.fabric.io/ios/changelog.html#fabric
pod 'Crashlytics',              '3.7.0'   # https://docs.fabric.io/ios/changelog.html#crashlytics
pod 'Digits',                   '1.15.1'  # https://docs.fabric.io/ios/changelog.html#digits
pod 'TwitterCore',              '1.15.1'  # https://docs.fabric.io/ios/changelog.html#twittercore
pod 'Parse',                    '1.13.0'  # https://github.com/ParsePlatform/Parse-SDK-iOS-OSX/releases
pod 'ParseUI',                  '1.2.0'   # https://github.com/ParsePlatform/ParseUI-iOS/releases
pod 'libPhoneNumber-iOS',       '0.8.10'  # https://github.com/iziz/libPhoneNumber-iOS/wiki/Update-Log and https://github.com/iziz/libPhoneNumber-iOS/releases
pod 'DateTools',                '1.7.0'   # https://github.com/MatthewYork/DateTools/releases
pod 'CSGrowingTextView',        '1.0.5'   # https://github.com/cloverstudio/CSGrowingTextView/releases
pod 'NSString+RemoveEmoji',     '0.1.1'   # https://github.com/woxtu/NSString-RemoveEmoji/releases
pod 'LLSimpleCamera',           '4.1.0',  :inhibit_warnings => true   # https://github.com/omergul123/LLSimpleCamera/releases
pod 'Mixpanel',                 '2.9.3'   # https://github.com/mixpanel/mixpanel-iphone/releases
pod 'Branch',                   '0.11.15',:inhibit_warnings => true   # https://github.com/BranchMetrics/iOS-Deferred-Deep-Linking-SDK/releases
pod 'PureLayout',               '3.0.1'   # https://github.com/PureLayout/PureLayout/releases
pod 'LGAlertView',              '2.0.13'  # https://github.com/Friend-LGA/LGAlertView/releases
pod 'TTTAttributedLabel',       '1.13.4', :inhibit_warnings => true   # https://github.com/TTTAttributedLabel/TTTAttributedLabel/releases
pod 'ParseLiveQuery',           '1.0.1'   # https://github.com/ParsePlatform/ParseLiveQuery-iOS-OSX/releases

#post_install do |installer|
#    `find Pods -regex 'Pods/Parse.*\\.h' -print0 | xargs -0 sed -i '' 's/\\(<\\)Parse\\/\\(.*\\)\\(>\\)/\\"\\2\\"/'`
#    `find Pods -regex 'Pods/Bolts.*\\.h' -print0 | xargs -0 sed -i '' 's/\\(<\\)Bolts\\/\\(.*\\)\\(>\\)/\\"\\2\\"/'`
#    `find Pods -regex 'Pods/Parse.*\\.h' -print0 | xargs -0 sed -i '' 's/#import <Bolts\\/BFTask.h>/#import <Bolts\\/Bolts-umbrella.h>/'`
#    `find Pods -regex 'Pods/SocketRocket.*\\.h' -print0 | xargs -0 sed -i '' 's/\\(<\\)SocketRocket\\/\\(.*\\)\\(>\\)/\\"\\2\\"/'`
#end
```

### Error

```
NoMethodError - undefined method `include?' for nil:NilClass
/Library/Ruby/Gems/2.0.0/gems/cocoapods-deintegrate-1.0.0.rc.1/lib/cocoapods/deintegrator.rb:57:in `block in deintegrate_shell_script_phase'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-deintegrate-1.0.0.rc.1/lib/cocoapods/deintegrator.rb:56:in `select'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-deintegrate-1.0.0.rc.1/lib/cocoapods/deintegrator.rb:56:in `deintegrate_shell_script_phase'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-deintegrate-1.0.0.rc.1/lib/cocoapods/deintegrator.rb:22:in `block in deintegrate_target'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/lib/cocoapods/user_interface.rb:63:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-deintegrate-1.0.0.rc.1/lib/cocoapods/deintegrator.rb:21:in `deintegrate_target'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-deintegrate-1.0.0.rc.1/lib/cocoapods/deintegrator.rb:8:in `block (2 levels) in deintegrate_project'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-deintegrate-1.0.0.rc.1/lib/cocoapods/deintegrator.rb:7:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-deintegrate-1.0.0.rc.1/lib/cocoapods/deintegrator.rb:7:in `block in deintegrate_project'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/lib/cocoapods/user_interface.rb:63:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-deintegrate-1.0.0.rc.1/lib/cocoapods/deintegrator.rb:6:in `deintegrate_project'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/lib/cocoapods/installer.rb:499:in `block (3 levels) in deintegrate_if_different_major_version'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/lib/cocoapods/config.rb:39:in `with_changes'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/lib/cocoapods/installer.rb:499:in `block (2 levels) in deintegrate_if_different_major_version'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/lib/cocoapods/installer.rb:498:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/lib/cocoapods/installer.rb:498:in `block in deintegrate_if_different_major_version'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/lib/cocoapods/user_interface.rb:63:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/lib/cocoapods/installer.rb:495:in `deintegrate_if_different_major_version'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/lib/cocoapods/installer.rb:132:in `block in prepare'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/lib/cocoapods/user_interface.rb:144:in `message'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/lib/cocoapods/installer.rb:131:in `prepare'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/lib/cocoapods/installer.rb:113:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/lib/cocoapods/command/install.rb:37:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-1.0.0.rc.1/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/lib/cocoapods/command.rb:50:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0.rc.1/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```

**Comment by segiddins** [comment]
@chadpav that's a totally different issue, please open an issue on CocoaPods/cocoapods-deintegrate with a project we can use to reproduce. Thanks!

**Comment by dadalar** [comment]
@chadpav you can manually delete Pods directory and Podfile.lock file, then run pod install again.

**Comment by iblacksun** [comment]
@chadpav @dadalar I had same error and still  after delete Pods directory and Podfile.lock file.

**Comment by chadpav** [comment]
@dadalar @iblacksun I ended up resolving my issue but I admit I changed a couple of things at once and am not sure which one did it.

Deleting the pods director did not help. @matrinox was on the right track as the stack trace pointed to a missing tag in the project file for SDKRoot.  In my project settings this appeared correct but something I did added this value into a recent commit.  Maybe change it and then change it back? There seem to be several things in Xcode that aren't explicitly set in source but if you change the value and then change it back it gets written to disk then.

It sounds like this code is depending on a value that doesn't exist for everyone. Possibly only if your project stared from a prior version of Xcode (which could explain why a default new project seems to work fine).
